testsuite/gnat2goto/tests/arrays_concat/arrays_concat.adb

```diff
@@ -0,0 +1,8 @@
+procedure Arrays_Concat is
+   type Arr is array (1..4) of Integer;
+   Actual : Arr := (1,2,3,4);
+begin
+
+   Actual (1 .. 4) := Actual (3 .. 4) & Actual (1 .. 2);
+   pragma Assert (Actual(2)=4);
+end Arrays_Concat;
```

Comment:
As I understand it the new array assignment function generalises not just to self assignment but also
A = B + C 
if (A has size 4) and B and C have size two. 

You do the edge case of self assignment - that's essential (i.e. the memcpy temp array required)
But I'd find it helpful to see *in the test* that the general case is also covered. 

To someone, in years gone by, looking at these tests in isolation the thought might occur:
"Ok - I see this covers the edge case - do I know for sure that it works with the general case?".

Reply:
Hmm, I thought about the (potentially) overlapping case as the general one, but I can add a test for `A=B&C`.